src/main/java/taxiapp/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,43 @@
+package taxiapp.dao;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import taxiapp.db.Storage;
+import taxiapp.lib.Dao;
+import taxiapp.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return Storage.manufacturers.get(Storage.manufacturers.size() - 1);
```

Comment:
```suggestion
        return manufacturer;
```
it should be with id already at this point